autoscheduler/scraper/tests/api_tests.py

```diff
@@ -0,0 +1,253 @@
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(expected, response)
+
+    def test_api_course_serializer_gives_expected_output(self):
+        """ Tests that the course serializer yields the correct data """
+        # Arrange
+        expected = {'title': 'Introduction to Computing', 'credit_hours': 3}
+
+        # Act
+        serializer = CourseSerializer(self.courses[0])
+
+        # Assert
+        self.assertEqual(expected, serializer.data)
+
+    def test_api_course_serializer_handles_null_credit_hours(self):
+        """ Tests that the course serializer correctly handles null credit_hours """
+        # Arrange
+        expected = {'title': 'Sports Law', 'credit_hours': None}
+
+        # Act
+        serializer = CourseSerializer(self.courses[4])
+
+        # Assert
+        self.assertEqual(expected, serializer.data)
+
+    def test_api_course_gives_valid_response_csce(self):
+        """ Tests that /api/course?dept=CSCE&course_num=181&term=201931 gives the
+            correct output
+        """
+        # Arrange
+        expected = CourseSerializer(self.courses[0])
+        data = {'dept': 'CSCE', 'course_num': '181', 'term': '201931'}
+
+        # Act
+        response = self.client.get('/api/course', data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected.data)
+
+    def test_api_course_gives_valid_response_law(self):
+        """ Tests that /api/course?dept=LAW&course_num=7500S&term=202031 gives the
+            correct output (verifies API can handle null credit_hours)
+        """
+        # Arrange
+        expected = CourseSerializer(self.courses[4])
+        data = {'dept': 'LAW', 'course_num': '7500S', 'term': '202031'}
+
+        # Act
+        response = self.client.get('/api/course', data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected.data)
+
+    def test_api_section_serializer_gives_expected_output(self):
+        """ Tests that the section serializer yields the correct data """
+        # Arrange
+        first_start = time(11, 30)
+        first_end = time(12, 20)
+        second_start = time(9, 10)
+        second_end = time(10)
+        meeting_days = [True] * 7
+        expected = {
+            'crn': 12345,
+            'instructor_gpa': None,
+            'instructor_name': 'Akash Tyagi',
+            'honors_only': False,
+            'meetings': {
+                '10': {
+                    'days': meeting_days,
+                    'start_time': first_start,
+                    'end_time': first_end,
+                    'type': 'LEC',
+                },
+                '11': {
+                    'days': meeting_days,
+                    'start_time': second_start,
+                    'end_time': second_end,
+                    'type': 'LEC',
+                },
```

Comment:
Honestly I think meetings should be an array of meetings, as the ID's for them really aren't relevant (since they're just the section ID + an integer count for which meeting it is)